### DISCUSSION TITLE
Trigger re-download of remote servers custom emoji

### DISCUSSION THREAD

**Discussion by jbfriedrich** [mastodon/mastodon#22434] [Q&A]
Hey friends,

I had a little SNAFU when I tried to migrate my cache directory to S3. I know what I did wrong, so I will try again in the future without messing it up again. So I lost the whole cache. I re-uploaded my local custom emoji, I have run `RAILS_ENV=production bin/tootctl media refresh --force --days 90` and `RAILS_ENV=production bin/tootctl accounts --all`, but I am still missing the emoji from the remote servers. 

I tried to delete remote emoji and hoped they would rebuild on their own, but so far to no avail. How can I trigger ad re-download/sync/download of all servers I am federating with? Can anyone point me into the right direction, please?

Thanks,
-Jason

**Comment by karimlevallois** [comment]
I would also like to know how to re-download custom emojis that my instance has federated with as I accidentally deleted a number of them.

**Comment by Tslmuun** [comment]
https://xxx.xxx.xx/admin/custom_emojis?remote=1
Remove all torn remote emoji. it will re-download.

**Reply by jbfriedrich** [reply]
Yeah, sure whenever someone from that remote instance uses an emoji in a completely new post, it will be downloaded.

It the meantime, all other emoji are broken in profiles and messages. And when I scroll to past messages, the emoji are not loaded either.

Is there no way to trigger a redownload of all emoji from foo.com? Or from all servers my instance federates with? (of course not all at once, staggered approach for 48 hours or so would be fine).

**Reply by Tslmuun** [reply]
I don't know how to trigger foo.com's emoji to re-download
Wait until all remote emojis are re-downloaded, and the emojis in the historical post will be reloaded.

**Comment by blacklight** [comment]
For those like me - who migrated a Mastodon instance, saw a cache directory and though "oh, it's just a cache directory - I will remove it and the new version of the software should repopulate it", and eventually lost all the emojis: I have made a [workaround script](https://gist.github.com/BlackLight/5f08fdfb629d6c910a1142a302f320d8).

I had about 40k missing emojis after the migration, so just going after them one by one in the admin page and removing those that were missing was unfeasible.

This script will take care of:

- Fetching all the custom emojis
- Check all those that don't have associated cache files
- Remove them from the database

IMHO this should probably be a feature of Mastodon itself - `purge_cache` or something similar? Some cached content is missing, and the application can't spot that the content is missing, and can't even be forced to check and redownload the missing content: this sounds like a bug to me.

**Reply by cham2019** [reply]
Thanks for the script.  I used it but it only deleted the broken references in the DB.  
It didn't fetch or re-sync any of the broken for actual posts loaded on my server. Maybe I did something wrong.
![rty657](https://user-images.githubusercontent.com/48077863/234871636-8ce4ee43-17d2-4e7d-aa54-b98966b41a7f.jpg)

**Comment by winguse** [comment]
in case anyone find this again, I found we can now do this by 

```tootctl emoji purge --remote-only``` 

see https://github.com/mastodon/mastodon/issues/12804
